test/unit-test/core_mqtt_utest.c

```diff
@@ -1949,17 +2011,21 @@ void test_MQTT_ProcessLoop_Timer_Overflow( void )
     MQTTPublishState_t publishState = MQTTPubAckSend;
     MQTTPublishState_t ackState = MQTTPublishDone;
     uint8_t i = 0;
-    uint8_t numIterations = ( MQTT_TIMER_OVERFLOW_TIMEOUT_MS / MQTT_TIMER_CALLS_PER_ITERATION ) + 1;
+    uint8_t numIterations = ( ( MQTT_TIMER_OVERFLOW_TIMEOUT_MS % MQTT_TIMER_CALLS_PER_ITERATION ) == 0 ) ?
+                            ( MQTT_TIMER_OVERFLOW_TIMEOUT_MS / MQTT_TIMER_CALLS_PER_ITERATION ) :
+                            ( ( MQTT_TIMER_OVERFLOW_TIMEOUT_MS / MQTT_TIMER_CALLS_PER_ITERATION ) + 1 );
+
     uint32_t expectedFinalTime;
 
     setupTransportInterface( &transport );
+    setupNetworkBuffer( &networkBuffer );
 
     networkBuffer.size = 1000;
     incomingPacket.type = MQTT_PACKET_TYPE_PUBLISH;
     incomingPacket.remainingLength = MQTT_SAMPLE_REMAINING_LENGTH;
 
     globalEntryTime = UINT32_MAX - MQTT_OVERFLOW_OFFSET;
-    expectedFinalTime = MQTT_TIMER_CALLS_PER_ITERATION * numIterations - MQTT_OVERFLOW_OFFSET;
+    expectedFinalTime = globalEntryTime + ( numIterations * MQTT_TIMER_CALLS_PER_ITERATION ) + 1;
```

Comment:
Nit: Why this change? I liked the way it looked before, where we calculate the expected time independently. If we're testing if an overflow calculation works as expected, it looks a little weird to me that we're doing the exact same overflow calculation here (I know since they're unsigned it's not really an "overflow" and addition is well-defined, and that these two expressions will always be identical for uint32_t, but you know what I mean).

Reply:
This is correcting the calculation of the expected time counter after the call to `MQTT_ProcessLoop`. The `+1` is for the call made to `getTime` before all iterations of the loop in the API.

Reply:
What's the correction? This value is the same as before isn't it? 
`(UINT32_MAX - MQTT_OVERFLOW_OFFSET) + (numIterations * MQTT_TIMER_CALLS_PER_ITERATION ) + 1` is the same as 
`((0) - MQTT_OVERFLOW_OFFSET) + (numIterations * MQTT_TIMER_CALLS_PER_ITERATION)` or
`MQTT_TIMER_CALLS_PER_ITERATION * numIterations - MQTT_OVERFLOW_OFFSET`

Also, basing `expectedFinalTime` off of `globalEntryTime` results in a calculation that always yields the final value, regardless if the counter rolls over back to 0 or not, which is what this test seems to be testing (e.g. If someone accidentally changed the value of `globalEntryTime` so there's no longer a rollover, this test would still pass).

Ultimately it doesn't matter, since the value is the same as it was before, but that's why this change looked weird to me

Reply:
>Also, basing expectedFinalTime off of globalEntryTime results in a calculation that always yields the final value, regardless if the counter rolls over back to 0 or not,

This is the main motivation from a resiliency and readability perspective.

Reply:
I was saying it's less resilient since before, accidentally setting `globalEntryTime` incorrectly could be caught due to the test failing. Now, due to how `expectedFinalTime` is calculated from `globalEntryTime`, the test will always pass regardless of the initial value of the entry time